Demo/Demo/FeatureViewControllers/SwiftUICardDemoView.swift

```diff
@@ -0,0 +1,114 @@
+struct SwiftUICardDemo: View {
+
+    // MARK: Variables
+
+    @State private var cardNumberText: String = ""
+    @State private var expirationDateText: String = ""
+    @State private var cvvText: String = ""
+
+    @StateObject var baseViewModel = BaseViewModel()
+
+    private let cardFormatter = CardFormatter()
+
+    // MARK: Views
+
+    var body: some View {
+        ZStack {
+            FeatureBaseViewControllerRepresentable(baseViewModel: baseViewModel)
+            VStack(spacing: 16) {
+                FloatingLabelTextField(placeholder: "Card Number", text: $cardNumberText)
+                    .onChange(of: cardNumberText) { newValue in
+                        formatFieldWith(newValue, field: .cardNumber)
+                    }
+                FloatingLabelTextField(placeholder: "Expiration Date", text: $expirationDateText)
+                    .onChange(of: expirationDateText) { newValue in
+                        formatFieldWith(newValue, field: .expirationDate)
+                    }
+                FloatingLabelTextField(placeholder: "CVV", text: $cvvText)
+                    .onChange(of: cvvText) { newValue in
+                        formatFieldWith(newValue, field: .cvv)
+                    }
+                Button("\(DemoSettings.intent.rawValue.capitalized) Order") {
+                    guard let card = createCard(), let orderID = baseViewModel.orderID else {
+                        baseViewModel.updateTitle("Failed: missing card / orderID.")
+                        return
+                    }
+
+                    checkoutWithCard(card, orderID: orderID)
+                }
+                .foregroundColor(.white)
+                .padding()
+                .frame(maxWidth: .infinity)
+                .background(.blue)
+                .cornerRadius(10)
+                .disabled(!(cvvText.count >= 3 && cardNumberText.count >= 17 && expirationDateText.count == 7))
+            }
+            .padding(.horizontal, 16)
+        }
+    }
+
+    // MARK: Functions
+
+    private func formatFieldWith(_ text: String, field: Fields) {
+        switch field {
+        case .cardNumber:
+            let cleanedText = text.replacingOccurrences(of: " ", with: "")
+            cardNumberText = String(cleanedText.prefix(CardType.unknown.getCardType(cardNumberText).maxLength))
+            cardNumberText = cardNumberText.filter { ("0"..."9").contains($0) }
+            cardNumberText = cardFormatter.formatCardNumber(cardNumberText)
+        case .expirationDate:
+            let cleanedText = text.replacingOccurrences(of: " / ", with: "")
+            expirationDateText = String(cleanedText.prefix(4))
+            expirationDateText = expirationDateText.filter { ("0"..."9").contains($0) }
+            expirationDateText = cardFormatter.formatExpirationDate(expirationDateText)
+
+        case .cvv:
+            cvvText = String(text.prefix(4))
+            cvvText = cvvText.filter { ("0"..."9").contains($0) }
```

Comment:
Can we move `formatCardNumber` and `formatExpirationDate` functions from `CardDemoViewController` to maybe `CardFormatter`, and then reuse them here instead of writing new? :smile:

Reply:
That's a good call - I'll move this logic in there to share across the 2 interfaces! 😄